lucene/core/src/java/org/apache/lucene/search/TopFieldCollector.java

```diff
@@ -68,6 +68,18 @@ public void setScorer(Scorable scorer) throws IOException {
   }
 
   static boolean canEarlyTerminate(Sort searchSort, Sort indexSort) {
+    return canEarlyTerminateOnDocId(searchSort, indexSort) ||
+           canEarlyTerminateOnPrefix(searchSort, indexSort);
+  }
+
+  private static boolean canEarlyTerminateOnDocId(Sort searchSort, Sort indexSort) {
```

Comment:
> since docs are visited in doc Id order, if compare is 0, it means
> this document is largest than anything else in the queue, and
> therefore not competitive.

I'm not at ease with this change because this comment states that docs are visited in doc id order. Therefore did you mean we should completely remove the `canEarlyTerminateOnPrefix` early termination? Thanks.

Reply:
Today early termination is activated if the search sort is a prefix of the index sort. With this pr we add a new case for early termination that should be activated when the search sort uses SortField.DOC. The early termination is possible in this case because docs are visited in doc id order so the first visited document is the also the best document, the second visited is the second best document, and so on.